src/daq/src/WaveformAnalysisSinc.cc

```diff
@@ -0,0 +1,110 @@
+#include <TF1.h>
+#include <TH1D.h>
+#include <TMath.h>
+
+#include <RAT/Log.hh>
+#include <RAT/WaveformAnalysisSinc.hh>
+
+#include "RAT/DS/DigitPMT.hh"
+#include "RAT/DS/WaveformAnalysisResult.hh"
+#include "RAT/WaveformUtil.hh"
+
+namespace RAT {
+void WaveformAnalysisSinc::Configure(const std::string& config_name) {
+  try {
+    fDigit = DB::Get()->GetLink("DIGITIZER_ANALYSIS", config_name);
+    fFitWindowLow = fDigit->GetD("fit_window_low");
+    fFitWindowHigh = fDigit->GetD("fit_window_high");
+  } catch (DBNotFoundError) {
+    RAT::Log::Die("WaveformAnalysisSinc: Unable to find analysis parameters.");
+  }
+}
+
+void WaveformAnalysisSinc::SetD(std::string param, double value) {
+  if (param == "fit_window_low") {
+    fFitWindowLow = value;
+  } else if (param == "fit_window_high") {
+    fFitWindowHigh = value;
+  } else {
+    throw Processor::ParamUnknown(param);
+  }
+}
+
+void WaveformAnalysisSinc::DoAnalysis(DS::DigitPMT* digitpmt, const std::vector<UShort_t>& digitWfm) {
+  double pedestal = digitpmt->GetPedestal();
+  if (pedestal == -9999) {
+    RAT::Log::Die("WaveformAnalysisSinc: Pedestal is invalid! Did you run WaveformPrep first?");
+  }
+  // Convert from ADC to mV
+  std::vector<double> voltWfm = WaveformUtil::ADCtoVoltage(digitWfm, fVoltageRes, pedestal = pedestal);
+  fDigitTimeInWindow = digitpmt->GetDigitizedTimeNoOffset();
+
+  if (std::isinf(fDigitTimeInWindow) || fDigitTimeInWindow < 0) {
+    return;
+  }
+
+  InterpolateWaveform(voltWfm);
+
+  DS::WaveformAnalysisResult* fit_result = digitpmt->GetOrCreateWaveformAnalysisResult("Sinc");
+  fit_result->AddPE(fFitTime, fFitCharge, {{"peak", fFitPeak}});
+}
+
+std::vector<double> WaveformAnalysisSinc::convolve_wfm(const std::vector<double>& wfm,
+                                                       const std::vector<double>& kernel) {
+  int n = wfm.size();
+  int m = kernel.size();
+  std::vector<double> result(n + m - 1, 0.0);  // Size of the output
+  // Perform the convolution
+  for (int i = 0; i < n; ++i) {
+    for (int j = 0; j < m; ++j) {
+      result[i + j] += wfm[i] * kernel[j];
+    }
+  }
+  return result;
+}
+
+void WaveformAnalysisSinc::InterpolateWaveform(const std::vector<double>& voltWfm) {
+  // Fit range
+  double bf = fDigitTimeInWindow - fFitWindowLow;
+  double tf = fDigitTimeInWindow + fFitWindowHigh;
+
+  // Check the fit range is within the digitizer window
+  bf = (bf > 0) ? bf : 0;
+  tf = (tf > voltWfm.size() * fTimeStep) ? voltWfm.size() * fTimeStep : tf;
+
+  // Get samples values within the fit range
+  std::vector<double> fit_wfm;
+  int sample_low = static_cast<int>(floor(bf / fTimeStep));
+  int sample_high = static_cast<int>(floor(tf / fTimeStep));
+  sample_high = std::min(static_cast<int>(voltWfm.size()) - 1, sample_high);
+  for (int j = sample_low; j < sample_high + 1; j++) {
+    fit_wfm.push_back(voltWfm[j]);
+  }
+
+  // Calculating tapered sinc kernel
+  std::vector<double> tsinc_kernel;
```

Comment:
I believe this kernel calculation is independent of the input waveform parameters. Meaning that the kernel calculation should be placed in a static function, and should only be computed once, not during every single `DoAnalysis` call. 

Furthermore, currently it seems all kernel parameters are hard-coded. I would suggest one of the following two changes:
1. Expose relevant parameters (for example, ones that adjust the Fourier-Domain bandwidth of the sinc kernel) as a ratdb table entry
2. convert this kernel calculation to a pre-calculated array, placing it in a ratdb entry. 
While option 2 may seem inferior at first glance,  it would actually make this fitter a general "convolution" analyzer. I am not sure how useful this would be since the charge integration is quite simple, but perhaps people would want to run kernels with better time-domain behaviors.